packages/manager/apps/pci-load-balancer/src/api/hook/useGateways.ts

```diff
@@ -20,3 +29,61 @@ export const useGetSubnetGateways = (
     enabled: !!projectId && !!regionName && !!subnetId,
     throwOnError: true,
   });
+
+export const useRegionPlans = (regionName: string) => {
+  const { data: project } = useProject();
+  const { data } = useProductAvailability(project.project_id, {
+    addonFamily: 'gateway',
+  });
+
+  return useMemo(() => {
+    const plans = data?.plans.filter(({ code }) => code.includes('hour')) || [];
+    const regionPlans =
+      plans.filter((plan) =>
+        plan.regions.some((region) => region.name === regionName),
+      ) || [];
+
+    return regionPlans.map((item) => item.code);
+  }, [data, regionName]);
+};
+
+const getGatewaySize = (addon: TAddon) => addon.product.split('-').pop();
+
+type TGatewayCatalog = {
+  size: string;
+  price: number;
+};
+
+const getSmallestGateway = (
+  catalog: TCatalog,
+  plans: string[],
+): TGatewayCatalog => {
+  const sizeOrder = ['s', 'm', 'l', 'xl', '2xl'];
+
+  const gateways = catalog.addons
+    .filter((addon) => plans.includes(addon.planCode))
+    .sort((a, b) => {
+      const sizeA = getGatewaySize(a);
+      const sizeB = getGatewaySize(b);
+
+      return sizeOrder.indexOf(sizeA) - sizeOrder.indexOf(sizeB);
+    });
```

Comment:
Prefer immutability when handling sorts: 

```suggestion
  const gateways = catalog.addons
    .filter((addon) => plans.includes(addon.planCode))
    .slice()
    .sort((a, b) => {
      const sizeA = getGatewaySize(a);
      const sizeB = getGatewaySize(b);

      return sizeOrder.indexOf(sizeA) - sizeOrder.indexOf(sizeB);
    });
```
or using spread operator :)